Apache Pulsar: Document support for new auth modes (#977)
kedacore/keda-docs: content/docs/2.9/scalers/pulsar.md

- `subscription` - Name of the topic subscription
- `msgBacklogThreshold` - Average target value to trigger scaling actions. (default: 10)
- `activationMsgBacklogThreshold` - Target value for activating the scaler. Learn more about activation [here](./../concepts/scaling-deployments.md#activating-and-scaling-thresholds).(Default: `0`, Optional)

> - `authModes` - a comma separated list of authentication modes to use. Default `""`. Valid options are `bearer`, `tls`, and `basic`. Note that `tls,bearer` or `tls,basic` are valid combinations and would indicate mutual TLS to secure the connection and then `bearer` or `basic` headers should be added to the HTTP request.

Reviewer: ```suggestion
- `authModes` - a comma separated list of authentication modes to use. (Values: `bearer`, `tls`,`basic`, Default: `""`, Optional, `tls,bearer` or `tls,basic` are valid combinations and would indicate mutual TLS to secure the connection and then `bearer` or `basic` headers should be added to the HTTP request)
```

Reviewer 2: @JorTurFer Shouldn't the values/default be at the end?

Reviewer: I have checked it, the style is 

- name - Description. (Values: x, y, z, Default: y, Optional, Extra Info)

So I think that my suggestion is correct

### Authentication Parameters

>   The `authModes` are specified on the `metadata` field of the `ScaledObject`, as shown above. The associated configuration parameters are specified in the `TriggerAuthentication` spec.  

Reviewer 2: ```suggestion
The authentication is defined in `authModes`. The associated configuration parameters are specified in the `TriggerAuthentication` spec.  
```

**Bearer Auth**

When configuring Bearer Authentication (Token Auth), configure the following:

- `bearerToken`: This token will be sent as a header in the form `Authorization: Bearer <token>`. 

**Basic Auth**
When configuring Basic Authentication, configure the following:

- `username`: the username
- `password`: the password (optional)

**TLS:**

When configuring mutual TLS authentication, configure the following:

- `ca`: The trusted root Certificate authority used to validate the server's certificate.
- `cert`: Certificate for client authentication.
- `key`: Key for client authentication. 

### TLS with custom CA Certificates

When configuring a trusted root CA that is not well known, it is sufficient to specify the `ca` field on the `TriggerAuthentication` resource. See [Bearer Token with TLS via custom trusted CA Certificate](#bearer-token-with-tls-via-custom-trusted-ca-certificate) for an example.

> Before 2.9.0, it was necessary to configure `tls: enable` on the `ScaledObject`. That was removed in a backwards compatible way, so you can remove that field now.

Reviewer: I'm not sure if we should specify this here because this is 2.9 indeed, so changes are clear. We can/should add a deprecation message in the log, but I'm not sure if here is the best place.
@kedacore/keda-docs ?

Reviewer 3: I think that it is okay to keep it here, probably marked as 
```
> **Note:**
>
```

I am not sure if we want to have a deprecation message in the log? as the config will still work, it won't be changed in the future

Reviewer: do we want to support the old `tls` approach having the new one?

Reviewer 3: @JorTurFer I am not sure I follow. @michaeljmarshall mentioned that the change is done in a backwards compatible way. Ie -> before it was needed to specify `tls: enable`, now you don't have to do so, but you can keep the setting there. No change in the functionality in the end.

Reviewer: Maybe I misunderstood the change, but I supposed this will be the new "desired" way for the scaler. Am I right? @michaeljmarshall ?

Reviewer: @kedacore/keda-docs , we should merge this soon because the feature was merged by error before having the docs ready.
Do you have any thoughts you'd like to share?

Reviewer 2: I don't see what the problem is with this PR to be honest.

The security fix by @michaeljmarshall seems unrelated to this and does not provide any configuration options anyway; but I might be missing something.

Reviewer 3: @JorTurFer @tomkerkhove so what about this?

Reviewer 2: Resolving
